src/Microsoft.Health.Fhir.CosmosDb.Initialization/Features/Storage/DataPlaneCollectionSetup.cs

```diff
@@ -0,0 +1,149 @@
+            }
+            catch (Exception ex)
+            {
+                LogLevel logLevel = ex is RequestRateExceededException ? LogLevel.Warning : LogLevel.Critical;
+                _logger.Log(logLevel, ex, "Collections initialization failed");
+                throw;
+            }
+        }
+
+        public async Task UpdateFhirCollectionSettings(Container container, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(container, nameof(container));
+
+            var thisVersion = await GetLatestCollectionVersion(container, cancellationToken);
+
+            if (thisVersion.Version < 2)
+            {
+                var containerResponse = await container.ReadContainerAsync(cancellationToken: cancellationToken);
+
+                // For more information on setting indexing policies refer to:
+                // https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-manage-indexing-policy
+                // It is no longer necessary to explicitly set the kind (range/hash)
+                containerResponse.Resource.IndexingPolicy.IncludedPaths.Clear();
+                containerResponse.Resource.IndexingPolicy.IncludedPaths.Add(new IncludedPath
+                {
+                    Path = "/*",
+                });
+
+                containerResponse.Resource.IndexingPolicy.ExcludedPaths.Clear();
+                containerResponse.Resource.IndexingPolicy.ExcludedPaths.Add(new ExcludedPath { Path = $"/{Constants.RawResource}/*", });
+                containerResponse.Resource.IndexingPolicy.ExcludedPaths.Add(new ExcludedPath { Path = $"/\"_etag\"/?", });
+
+                // Setting the DefaultTTL to -1 means that by default all documents in the collection will live forever
+                // but the Cosmos DB service should monitor this collection for documents that have overridden this default.
+                // See: https://docs.microsoft.com/en-us/azure/cosmos-db/time-to-live
+                containerResponse.Resource.DefaultTimeToLive = -1;
+
+                await container.ReplaceContainerAsync(containerResponse, cancellationToken: cancellationToken);
+
+                thisVersion.Version = CollectionSettingsVersion;
+                await container.UpsertItemAsync(thisVersion, new PartitionKey(thisVersion.PartitionKey), cancellationToken: cancellationToken);
+            }
+
+            // TODO:Need to Validate this logic
+           /* else if (thisVersion.Version < CollectionSettingsVersion)
+            {
+                await _updateSP.Execute(container.Scripts, cancellationToken);
+                thisVersion.Version = CollectionSettingsVersion;
+                await container.UpsertItemAsync(thisVersion, new PartitionKey(thisVersion.PartitionKey), cancellationToken: cancellationToken);
+            } */
+        }
+
+        private static async Task<CollectionVersion> GetLatestCollectionVersion(Container container, CancellationToken cancellationToken)
```

Comment:
Let's rename this method to "GetLatestCollectionVersion**Async**".

As a standard, all async methods should have the "Async" suffix.

Reply:
done